snap/naming/validate.go

```diff
@@ -163,3 +164,45 @@ func ValidateSocket(name string) error {
 	}
 	return nil
 }
+
+var errInvalidSecurityTag = errors.New("invalid security tag")
+
+// ValidateSecurityTag validates known variants of snap security tag.
+//
+// Two forms are recognised, one for apps and one for hooks. Other forms
+// are possible but are not handled here.
+//
+// TODO: handle the weird udev variant.
+func ValidateSecurityTag(tag string) error {
```

Comment:
How has this not existed before?!

Reply:
Ha, right? I guess we never had to do it from this side. We always looked at the tag from C.